packages/webapp/src/pages/Integrations/providerConfigKey/Endpoints/components/One.tsx

```diff
@@ -0,0 +1,241 @@
+
+    const [requestSnippet, responseSnippet] = useMemo(() => {
+        let req = '';
+        let res = '';
+
+        const activeEndpointIndex = flow.endpoints.findIndex((endpoint) => {
+            const obj = Object.entries(endpoint)[0];
+            return obj[0] === flow.endpoint.verb && obj[1] === flow.endpoint.path;
+        });
+        const outputModelName = Array.isArray(flow.returns) ? flow.returns[activeEndpointIndex] : flow.returns;
+        // This code is completely valid but webpack is complaining for some obscure reason
+        const outputModel = (flow.models as unknown as NangoModel[]).find((m) => m.name === outputModelName);
+
+        const providerConfigKey = integration.integration.unique_key;
+        const secretKey = environmentAndAccount!.environment.secret_key;
+
+        // Request
+        if (language === 'node') {
+            req =
+                flow.type === 'sync'
+                    ? nodeSyncSnippet({ modelName: outputModel!.name, secretKey, connectionId, providerConfigKey })
+                    : nodeActionSnippet({ actionName: flow.name, secretKey, connectionId, providerConfigKey, input: flow.input });
+        } else {
+            req = curlSnippet({
+                baseUrl,
+                endpoint: flow.endpoints[activeEndpointIndex],
+                secretKey,
+                connectionId,
+                providerConfigKey,
+                input: flow.input
+            });
+        }
+
+        // Response
+        if (flow.type === 'sync') {
+            res = outputModel ? getSyncResponse(outputModel) : 'no response';
+        } else {
+            res = outputModel ? modelToString(outputModel) : 'no response';
+        }
+
+        return [req, res];
+    }, [flow, language, baseUrl, environmentAndAccount, integration]);
+
+    const queryParams = useMemo(() => {
+        return flow.type === 'sync' ? syncDefaultQueryParams : null;
+    }, [flow]);
+
+    const body = useMemo(() => {
+        return flow.type === 'action' && flow.input ? flow.input : null;
+    }, [flow.input, flow.type]);
+
+    const metadata = useMemo(() => {
+        return flow.type === 'sync' && flow.input ? flow.input : null;
+    }, [flow.input, flow.type]);
+
+    return (
+        <div className="flex flex-col gap-10 text-white text-sm">
+            <header className="bg-active-gray flex gap-10 justify-between p-5">
+                <div className="flex flex-col gap-5">
+                    <h2>
+                        <HttpLabel {...flow.endpoint} size="xl" />{' '}
+                    </h2>
+                    <div>{flow.description}</div>
+                </div>
+                <div className="flex-shrink-0 content-center">
+                    <ScriptSettings flow={flow} integration={integration} />
+                </div>
+            </header>
+
+            {!flow.enabled && <Info variant="warning">This endpoint is disabled. To enable it, go to settings</Info>}
```

Comment:
Shouldn't clicking `settings` also open the drawer?

Reply:
yes I forgot to do that thanks 🙏🏻